### ISSUE TITLE
`TearsheetShell`, check for dialog feature flag

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#7300]
The `enable-experimental-focus-wrap-without-sentinels` feature flag is now an alias for the dialog feature flag in `@carbon/react`. Because the tearsheet shell uses this, all tearsheets are now by default using the new dialog component from `@carbon/react`.

This has caused some instances of failed tests because `HTMLDialog` is not included in jsdom. We should use `useFeatureFlag` and check if the dialog or focus sentinel flag is set, otherwise we should render the old way we had been before we introduced `enable-experimental-focus-wrap-without-sentinels` to the tearsheet.

**Comment by wkeese** [comment]
There are more problems too... if TearsheetShell renders using `<dialog>`, then confirmation modals (based on @carbon/react's Modal or ComposedModal) won't display over the Tearsheet unless they also use `<dialog>`.

But anyway, the solution is just removing the `<FeatureFlags enableExperimentalFocusWrapWithoutSentinels>` wrapper added in #5882?

**Comment by wkeese** [comment]
PS: @taylor has separated the two flags again in https://github.com/carbon-design-system/carbon/pull/19134.

But I still think it makes sense to remove the  `<FeatureFlags enableExperimentalFocusWrapWithoutSentinels>` wrapper from TearsheetShell.tsx.

The reason is that the `<FeatureFlags enableExperimentalFocusWrapWithoutSentinels>` hides any application-level `<FeatureFlags enableDialogElement >`.  So that precludes the application from getting the accessibility improvements of `enableDialogElement` when using Tearsheet.

**Comment by amal-k-joy** [comment]
We can continue to use enableExperimentalFocusWrapWithoutSentinels in the old Tearsheet and can explore the feasibility of using enableDialogElement in new composable Tearsheet.
